src/parent.ts

```diff
@@ -14,29 +14,51 @@ export class ParentClient<C = any> extends SharedClient<C> {
     }
 
     /**
-     * Request a channel with the child client. Must be called after child
+     * Request a channel with a child client in an iframe. Must be called after child
      * frame is fully loaded.
      */
     requestChannel<T>(frame: HTMLIFrameElement, context: T) {
-        this.url = new URL(frame.src);
-
         if (frame.contentWindow) {
-            const messageChannel = new MessageChannel();
+            this.handleRequstChannel<T>(
+                frame.contentWindow,
+                frame.src,
+                context
+            );
+        }
+    }
 
-            this.messagePort = messageChannel.port1;
+    /**
+     * Request a channel with the child client in a new tab or a new window opened with window.open().
+     * Must be called after the popup has fully loaded, if not blocked by a popup blocker.
+     */
+    requestChannelWithPopup<T>(targetWindow: Window, url: string, context: T) {
+        if (targetWindow.opener) {
+            this.handleRequstChannel<T>(targetWindow, url, context);
+        }
+    }
```

Comment:
Instead of making another method called `requestChannelWithPopup`, we could also make the original `requestChannel` method accept a `Window` instead of an `HTMLIFrameElement`. It's a breaking change but I left this in pre-release for a reason :) A suggestion, let me know what you think. I'm approving